bin/init.sh

```diff
@@ -80,6 +80,13 @@ function download_envoy_if_necessary () {
     echo "Copying $2 to $(dirname "$2")/${3}"
     cp -f "$2" "$(dirname "$2")/${3}"
     popd
+
+  else
+    # The Envoy binary already exists; this is the case when user overrides the
+    # ISTIO_ENVOY_LINUX_RELEASE_PATH variable to use their own Envoy.
+    # In this case, we must copy it to the release directory
+    mkdir -p "${ISTIO_ENVOY_LINUX_RELEASE_DIR}"
+    cp -f "$2" "${ISTIO_ENVOY_LINUX_RELEASE_DIR}"
```

Comment:
It seems that you are setting the `ISTIO_ENVOY_LINUX_RELEASE_PATH` and not setting `ISTIO_ENVOY_LINUX_RELEASE_DIR` before calling. In that case, `ISTIO_ENVOY_LINUX_RELEASE_PATH` will default to ${TARGET_OUT_LINUX}/release.  In your scenario, setting the DIR as the parent of the PATH (and maybe the we should do that if only the PATH is defined) does not require this change.

Reply:
I see your point. In this case, we would have to update https://github.com/istio/istio/blob/master/tools/docker.yaml#L32 as it hardcodes the path to the sidecar, assuming it's under `${TARGET_OUT_LINUX}/release`.

Reply:
> Or another option would be to always copy from the download or specified PATH to TARGET_OUT_LINUX. That seems strange.

Well, the more I think about it, that's probably not that strange since we copy to TARGET_OUT at the end of the script. And I'm curious why the copy from ISTIOENVOY_NATIVE_PATH isn't working.

Reply:
We rely on files being present on `$ISTIO_OUT/release`. This makes sure we copy the envoy binary when the user overrides it. Overriding `$ISTIO_ENVOY_LINUX_RELEASE_DIR` along with `$ISTIO_ENVOY_LINUX_RELEASE_PATH` might not work well.

In my case, for example, I have `ISTIO_ENVOY_LINUX_RELEASE_PATH=/envoy`. Thus `ISTIO_ENVOY_LINUX_RELEASE_DIR` had to be `/`, which clearly is not correct. So, not touching this DIR variable in general works just fine. With this PR, the only change users need to do to use a custom Envoy is to override a single variable, `ISTIO_ENVOY_LINUX_RELEASE_PATH`.